docs/configuration.md

```diff
@@ -0,0 +1,182 @@
+> 1. High Swap I/O Traffic:
+> * `averageSwapInPerSecond > maxAverageSwapInPagesPerSecond`
+> and<br>
+> `averageSwapOutPerSecond > maxAverageSwapOutPagesPerSecond`<br><br>
+> This condition is triggered when swap-related I/O traffic
+> is excessively high.
+> The default values for `maxAverageSwapInPagesPerSecond` and
+> `maxAverageSwapOutPagesPerSecond` are both set to 1000, averaged 
+> over a 30-second interval by default.
+> 2. High Memory Overcommitment:
+> * `maxMemoryOverCommitmentBytes < SwapUsedBytes - AvailableMemoryBytes`<br><br>
+> This condition is triggered when swap utilization is excessively high.
+> This depends on the `NODE_SWAP_SPACE` setting in the machine configuration from
+> step 3 or when the memory overcommitment ratio is too high.The default value for
+> `maxMemoryOverCommitment` is set to 500Mi.<br><br>
+>
+> `maxAverageSwapInPagesPerSecond`, `maxAverageSwapOutPagesPerSecond`, and 
+> `maxMemoryOverCommitment` can be adjusted by modifying the values of the 
+> `MAX_AVERAGE_SWAP_IN_PAGES_PER_SECOND`, `MAX_AVERAGE_SWAP_OUT_PAGES_PER_SECOND`, 
+> and `MEMORY_OVER_COMMITMENT_THRESHOLD` environment variables in the provided 
+> example, respectively. Additionally, the `AVERAGE_WINDOW_SIZE_SECONDS` environment 
+> variable determines the time frame for calculating the average.
+
+5. Deploy alerting rules according to the following
+   [example](../manifests/openshift/prometheus-rules.yaml) and
+   add the cluster-monitoring label to the wasp namespace.
+```console
+$ oc label namespace wasp openshift.io/cluster-monitoring="true"
+```
+
+6. Configure OpenShift Virtualization to use memory overcommit using
+
+   a. the OpenShift Console
+   b. the following [HCO example](../manifests/openshift/hco-set-memory-overcommit.yaml):
+
+```console
+$ oc patch --type=merge \
+  -f <../manifests/hco-set-memory-overcommit.yaml> \
+  --patch-file <../manifests/hco-set-memory-overcommit.yaml>
+```
+
+> [!NOTE]
+> After applying all configurations all `MachineConfigPool`
+> roll-outs have to complete before the feature is fully available.
+>
+>     $ oc wait mcp worker --for condition=Updated=True
+>
+
+### Upgrade path
+For users of wasp-agent v1.0, which lacks LimitedSwap and eviction support, here is the upgrade path:
+1. Adjust KubeletConfig:
+   If you have installed the KubeletConfiguration object from version v1.0, you need to update it. 
+   This can be done by applying the updated [KubeletConfiguration example](../manifests/openshift/kubelet-configuration-with-swap.yaml).
+```console
+$ oc apply -f <../manifests/openshift/kubelet-configuration-with-swap.yaml>
+```
+2. Update DaemonSet:
+   Apply the updated DaemonSet by using the provided [DaemonSet example](../manifests/openshift/ds.yaml).
+```console
+$ oc apply -f <../manifests/openshift/ds.yaml>
+```
+3. Update monitoring object:
+```console
+$ oc delete AlertingRule wasp-alerts -nopenshift-monitoring
+$ oc create -f <../manifests/openshift/prometheus-rules.yaml>
```

Comment:
```suggestion
$ oc delete PrometheusRule wasp-rules -n wasp
$ oc create -f <../manifests/openshift/prometheus-rule.yaml>
```

Reply:
If I understand correctly, in the upgrade path, we need to delete the old object, which is AlertingRule, and not PrometheusRule.
I updated the file name.

Reply:
It depends on the versions. The newly added alerts will be introduced in 4.17, right?
And what about this doc? If they will be introduced in the same version, I think my comment is correct, as this section should explain how to upgrade from the current version (which has the new alerts).

Reply:
The old object is AlertingRule only up to 4.16.